zml/buffer.zig

```diff
@@ -27,12 +27,29 @@ const log = std.log.scoped(.zml);
 /// * loading weights from disk directly to the `device zml.aio.loadBuffers`
 /// * can be created by calling `HostBuffer.toDevice(platform)`.
 pub const Buffer = struct {
+    pub const Shard = struct {
+        api: *const pjrt.Api,
```

Comment:
why do you need a copy of the api here ? can we have a buffer made of shards across different runtimes ?